### PR TITLE
build pulsar-adaptors with pulsar-2.8-snapshot

### LINKED ISSUE #5: Could provide more details about how to develop/build this project？
After Import this project into Intellij，got a lot of error messages like:

```
Could not find artifact org.apache.pulsar:pulsar:pom:2.7.0-SNAPSHOT
...
```

Seem like pulsar 2.7.0-SNAPSHOT is not in maven central pulic repo?

### PR DESCRIPTION
**Pull request by dlg99**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #5 

### Motivation

Tried building pulsar-adaptors locally with 2.8.0-SNAPSHOT.
Build failed.
Tried building with pulsar 2.7.1 release
Build fails.

Last pulsar-adaptors released with 2.6.x.
Some tests won't compile even with 2.7.1  release.

### Modifications

Made it build with locally built pulsar 2.8.0-SNAPSHOT (build as `mvn clean install -DskipTests`)
Updated pom files to include dependencies that project needed, same versions as pulsar IIRC. 
Fixed tests that failed after all that (due to changes in pulsar).  

One test needs fix in the pulsar, PR https://github.com/apache/pulsar/pull/10091

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

This change is already covered by existing tests, such as *(please describe tests)*.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (probably, should match pulsar's)

### Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: examples/flink/pom.xml
Comment (eolivelli) [inline examples/flink/pom.xml:40]: can we remove commented code ?

File: pulsar-client-kafka-compat/pulsar-client-kafka-tests/pom.xml
Comment (eolivelli) [inline pulsar-client-kafka-compat/pulsar-client-kafka-tests/pom.xml:49]: why this is not project.version ?

File: pulsar-client-kafka-compat/pulsar-client-kafka-tests/pom.xml
Comment (eolivelli) [inline pulsar-client-kafka-compat/pulsar-client-kafka-tests/pom.xml:47]: why this is not project.groupId ?

File: pulsar-flink/pom.xml
Comment (eolivelli) [inline pulsar-flink/pom.xml:206]: please remove commented code

**Review by eolivelli — COMMENTED** [review]
Nice work

@sijie please take a  look

**Comment by dlg99** [comment]
https://github.com/apache/pulsar/pull/10091 for pulsar-side fix

**Review by eolivelli — APPROVED** [review]
LGTM

**Review by eolivelli — COMMENTED** [review]
Very good

**Comment by eolivelli** [comment]
@merlimat @sijie can we commit this patch and unblock the repository?

**Comment by eolivelli** [comment]
@rdhabalia @merlimat can you help merging this patch ?

**Review by sijie — APPROVED** [review]

**Comment by sijie** [comment]
@codelipenghui Can you review it?

@eolivelli Can you request @codelipenghui for reviews for future PRs?

**Comment by eolivelli** [comment]
@sijie

> @eolivelli Can you request @codelipenghui for reviews for future PRs?

got it, thanks!

**Comment by sijie** [comment]
@eolivelli it would be good to make sure another review like from @codelipenghui before merging this, because I have already tagged him.

**Comment by eolivelli** [comment]
@sijie sorry.
 I thought that that your comment was for the next time

**Pull request merged**
